### ISSUE TITLE
Integration with Immich

### ISSUE THREAD

**Issue by seanmorley15** [seanmorley15/AdventureLog#196]
Import photos from Immich account. Possibly link using API keys from Immich.

**Comment by jeffklassen** [comment]
I agree with this! Also, while Google photos is not self-hosted, it might be a good idea to add Google photos integration.

**Comment by axgdcode** [comment]
II suggested to Immich the creation of customizable album types. I think photo albums generated from a GPX file (like Strava) would be relevant. 
https://github.com/immich-app/immich/discussions/6114

**Comment by nebriv** [comment]
Love this idea!

**Comment by floco** [comment]
Was looking for this functionality too and discovered https://dawarich.app/

**Comment by seanmorley15** [comment]
Hi everyone! I am super excited to announce the Immich integration is now available!
Right now it supports uploading photos to adventures from both smart search and albums. This is just the beginning of the integration. I foresee a future where basically no adventure editing is required, just link an Immich album and let AdventureLog do the work. Immich provides some very extensive API's that will make a tight integration possible. Sorry for the wait on this feature but I wanted to make sure it was done right :). Hope you all enjoy!!

**Comment by Vinod1452** [comment]
Hi  @seanmorley15 , this app is beautifully designed,  waiting for feature to link immich album for specific trip instead of uploading to Adventurelog, Also is there any possibility to add multiple locations to one adventure(example I visited 4-5 monuments in one day which were in same city, creating separate adventure for each will be difficult)

**Comment by seanmorley15** [comment]
@Vinod1452, thanks! Glad to hear you enjoy AdventureLog. Regarding linking albums, I think this would be a good feature for the future. There is a PR (#483) that will allow photos to be just displayed from Immich and not uploaded separately. For the multiple locations, adventures are supposed to be a single point on the map where a collection can be a whole trip or a folder for a day and you add the adventure locations to it. Let me know if this helps!

**Comment by albertusQuelle** [comment]
@seanmorley15 thank you so much for the idea and integration of immich. Is there any how to how to use it?
I have the API key from immich integrated in adventure log. 
But i dont know how to really use it. It also does not show me pictures, just broken previews. But it shows the folders.. 
Can i also upload date ranges of pictures?

<img width="1639" height="753" alt="Image" src="https://github.com/user-attachments/assets/292f66cc-ae18-400c-afee-e0a5bd462c8e" />
